org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.java

```diff
@@ -0,0 +1,489 @@
+/**
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xtext.ui.wizard.project;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.jface.dialogs.IMessageProvider;
+import org.eclipse.jface.fieldassist.ControlDecoration;
+import org.eclipse.jface.fieldassist.FieldDecoration;
+import org.eclipse.jface.fieldassist.FieldDecorationRegistry;
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.xtext.util.JUnitVersion;
+import org.eclipse.xtext.xbase.lib.Procedures.Procedure0;
+import org.eclipse.xtext.xbase.lib.Procedures.Procedure1;
+import org.eclipse.xtext.xtext.ui.internal.Activator;
+import org.eclipse.xtext.xtext.wizard.BuildSystem;
+import org.eclipse.xtext.xtext.wizard.LanguageServer;
+import org.eclipse.xtext.xtext.wizard.SourceLayout;
+import org.osgi.framework.Bundle;
+
+import com.google.common.collect.Iterables;
+
+public class AdvancedNewProjectPage extends WizardPage {
+
+	private Button createUiProject;
+
+	private Button createSDKProject;
+
+	private Button createP2Project;
+
+	private Button createWebProject;
+
+	private Button createIdeProject;
+
+	private Button createTestProject;
+
+	private Button junitVersion4;
+
+	private Button junitVersion5;
+
+	private Combo preferredBuildSystem;
+
+	private Combo createLanguageServer;
+
+	private Combo sourceLayout;
+
+	private Group createUiProjectSubGroup;
+
+	private StatusWidget statusWidget;
+
+	private boolean autoSelectIdeProject;
+
+	private boolean autoSelectSDKProject;
+
+	public AdvancedNewProjectPage(String pageName) {
+		super(pageName);
+		setTitle(Messages.AdvancedNewProjectPage_WindowTitle);
+		setDescription(Messages.AdvancedNewProjectPage_Description);
+	}
+
+	@Override
+	public void createControl(Composite parent) {
```

Comment:
@cdietrich @szarnekow @tivervac The size of this method and the nested lambda expressions made the task of translating to Java complicated. After migrating, I have verified this method twice; however, I may have overlooked something. I would appreciate it if you could do a thorough review of this method.

Reply:
LGTM except 1 comment. I feel your pain.